tests/std/tests/GH_002168_regex_overflow/test.cpp

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <regex>
+
+using namespace std;
+
+// GH-2168 <regex>: integer overflow on large backreference value
+int main() {
+    for (const char* regexString : {"\\3333333334", "\\2147483648"}) {

```

Comment:
Note that this is not testing what you think it's testing; this test succeeds on old code, because `\\2147483648` is not a backreference that exists in the string.